### PR TITLE
add url schema for API v2 routing

### PR DESCRIPTION
**Pull request by rizary**

### CODE REVIEW

**Comment by hvr** [comment]
@Rizary the good thing: the URL gets updated properly; the bad thing: if I enter the url, the page doesn't reflect the url state.

E.g. entering the url 

http://localhost:8080/#/package/titlecase/0.1.0.3/ghc-7.4.2@1495446092

manually should point to a build-failure of an older index-state, instead it shows the latest index-state, i.e. http://localhost:8080/#/package/titlecase@1500274475; also, it doesn't open up the logs-window for the 01.0.3/ghc-7.4.2 cell.

**Comment by rizary** [comment]
@hvr it seems that the issue come from purescript-routing. I will check on that.

**Comment by rizary** [comment]
I notice that your IRC is still broken? i have sent you couple of chat.

Re: Fail Deps button, what is come to my mind is whenever we click the button, drop-down menu appeared showing the list of `bfail` package with its name.

File: src-exe/Controller/Api.hs [hunk 1]
```diff
@@ -116,6 +116,7 @@ type ControllerApi m =
   :<|> "v2" :> "queue" :> Capture "pkgname" PkgN :> Capture "idxstate" PkgIdxTs :> Get '[JSON]  QEntryRow
   :<|> "v2" :> "queue" :> Capture "pkgname" PkgN :> Capture "idxstate" PkgIdxTs :> ReqBody '[JSON] QEntryUpd :> Put '[JSON] QEntryRow
   :<|> "v2" :> "queue" :> Capture "pkgname" PkgN :> Capture "idxstate" PkgIdxTs :> DeleteNoContent '[JSON] NoContent
+  :<|> "v2" :> "users" :> "name"                 :> Capture "username" UserName :> Get '[JSON] UserPkgs
```
Comment (hvr) [inline src-exe/Controller/Api.hs:119]: this results in `GET /api/v2/users/name/{username}`

we want instead:

 `GET /api/v2/users/{username}/packages`

So change this to:

```
:<|> "v2" :> "users" :> Capture "username" UserName :> "packages" :> Get '[JSON] UserPkgs
```

**Review by hvr — CHANGES_REQUESTED** [review]
:tada: for removing the old  v1  api javascript code! :-)

**Comment by hvr** [comment]
@Rizary most of this PR has been merged, except for https://github.com/haskell-CI/hackage-matrix-builder/pull/45/commits/437ac52d92e5fb93d4a0716ce696ad2381374145 for which I tried to cherry-pick into `master`, but it seems to be incomplete as I get the following compile failure:

```

[1/1 TypesDoNotUnify] /stuff3/work/GitHub/hackage-matrix-builder-3/src-ui.v2/src/Components/PageUser.purs:152:11

  152      st <- H.get
                 ^^^^^
  
  Could not match type
  
    ( packages :: Array (Tuple String Int)
    , user :: RemoteData Error (Array String)
    , withReports :: Boolean
    , initUser :: String
    )
  
  with type
  
    ( packages :: t2
    , userPkg :: t3
    , initUser :: String
    | t4
    )
  
  while trying to match type { initUser :: String
                             , user :: RemoteData Error (Array String)
                             , packages :: Array (Tuple String Int)
                             , withReports :: Boolean
                             }
    with type { initUser :: String
              , packages :: t2
              , userPkg :: t3
              | t4
              }

```

**Comment by rizary** [comment]
Closed as we move to Reflex-Dom.

**Pull request closed without merging**